src/java/org/apache/cassandra/index/sai/disk/v1/MemtableIndexWriter.java

```diff
@@ -110,7 +112,7 @@ public void complete(Stopwatch stopwatch) throws IOException
 
             try (MemtableTermsIterator terms = new MemtableTermsIterator(memtable.getMinTerm(), memtable.getMaxTerm(), iterator))
             {
-                long cellCount = flush(minKey, maxKey, indexContext.getValidator(), terms);
+                long cellCount = flush(minKey, maxKey, indexContext.getValidator(), terms, rowMapping.maxSSTableRowId);
```

Comment:
Just to clarify, we pass `maxSSTableRowId` here, but the argument we accept is `maxSegmentRowId`, because we're writing a single segment anyway?

Reply:
I've changed the `flush` method to take `maxSSTableRowId` because the `MemtableIndexWriter` is not really aware of segments as it only ever writes one. This value is passed to the `NumericIndexWriter` who interprets it as a `maxSegmentRowId` and we call `writeCompleteSegment` but I'm happy with that.